docs/mkdocs/docs/api/macros/json_diagnostic_positions.md

```diff
@@ -6,22 +6,36 @@
-
-`start_pos()` and `end_pos()` are only set if the JSON object was parsed using `parse()`. For all other cases, `std::string::npos` will be returned.
-
-Note that enabling this macro increases the size of every JSON value by two `std::size_t` fields and adds
-slight runtime overhead.
+When enabled, two new member functions [`start_pos()`](../basic_json/start_pos.md) and
+[`end_pos()`](../basic_json/end_pos.md) are added to [`basic_json`](../basic_json/index.md) values.
+[`start_pos()`](../basic_json/start_pos.md) returns the start position of that JSON value in the original JSON string
+the object was parsed from. Likewise, [`end_pos()`](../basic_json/end_pos.md) returns the end position of that JSON
+value in the original string the object was parsed from.
+
+[`start_pos()`](../basic_json/start_pos.md) returns the first character of a given value in the original JSON string,
+while [`end_pos()`](../basic_json/end_pos.md) returns the character _following_ the last character. For objects and
+arrays, the first and last characters correspond to the opening or closing braces/brackets, respectively. For primitive
+values, the first and last character represent the opening and closing quotes (strings) or the first and last character
+of the field's numerical or predefined value (`true`, `false`, `null`), respectively.
+
+| JSON type | return value [`start_pos()`](../basic_json/start_pos.md) | return value [`end_pos()`](../basic_json/end_pos.md) |
+|-----------|----------------------------------------------------------|------------------------------------------------------|
+| object    | position of the opening `{`                              | position after the closing `}`                       |
+| array     | position of the opening `[`                              | position after the closing `]`                       |
+| string    | position of the opening `"`                              | position after the closing `"`                       |
+| number    | position of the first character                          | position after the last character                    |
+| boolean   | position of `t` for `true` and `f` for `false`           | position after `e`                                   |
+| null      | position of `n`                                          | position after `l`                                   |
+
+Given the above, [`end_pos()`](../basic_json/end_pos.md)` - `[`start_pos()`](../basic_json/start_pos.md) for a JSON
+value provides the length of the parsed JSON string for that value, including the opening or closing braces, brackets,
+or quotes.
+
+[`start_pos()`](../basic_json/start_pos.md) and [`end_pos()`](../basic_json/end_pos.md) are only set if the JSON value
+was parsed using [`parse()`](../basic_json/parse.md). For all other cases,`std::string::npos` will be returned.
+
+Note that enabling this macro increases the size of every JSON value by two `std::size_t` fields and adds slight runtime
+overhead.
```

Comment:
...overhead to parsing, copying JSON value objects, and the generation of error messages for exceptions.  It also causes these values to be reported in those error messages.

Reply:
Done 37bb4d9c5d504ff0e8a9cd00cb33c84009ebcf95